src/megaclient.cpp

```diff
@@ -1825,9 +1787,10 @@ void MegaClient::exec()
                                 if (((*it)->state == SYNC_ACTIVE || (*it)->state == SYNC_INITIALSCAN)
                                  && !(*it)->dirnotify->notifyq[DirNotify::DIREVENTS].size()
                                  && !(*it)->dirnotify->notifyq[DirNotify::RETRY].size()
-                                 && !syncadding)
+                                 && !syncadding && syncuprequired && !syncnagleretry)
```

Comment:
I have used the same approach as in syncdown(). If any of the calls fails, the flag is set to keep retrying